### PR TITLE
Fix MSD docstring and accept AtomGroups

### LINKED ISSUE #2784: MSD docstring inconsistent with class inputs (select)
## Expected behavior ##

According to the docstring `select` defaults to `None` so not passing anything should just work.
https://github.com/MDAnalysis/mdanalysis/blob/333bb04e3a730278aa49ff4c6db372e800f05cd6/package/MDAnalysis/analysis/msd.py#L279-L281

## Actual behavior ##

`select` is a required positional argument.

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.analysis.msd import EinsteinMSD as MSD
from MDAnalysis.tests.datafiles import PSF, DCD

u = mda.Universe(PSF, DCD)

m = MSD(u)
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev0
- Which version of Python (`python -V`)? 3.7.7
- Which operating system? Ubuntu 20.04 LTS (WSL2)

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #2784 

Changes made in this Pull Request:
 - Fix the MSD docstring to be more accurate with respect to selections
 - Accept AtomGroups and reject UpdatingAtomGroups.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/msd.py [hunk 1]
```diff
@@ -340,7 +345,7 @@ def _single_frame(self):
         # shape of position array set here, use span in last dimension
         # from this point on
         self._position_array[self._frame_index] = (
-        self._atoms.positions[:, self._dim])
+            self._atoms.positions[:, self._dim])
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply package/MDAnalysis/analysis/msd.py:348]: im okay with it :)

File: package/MDAnalysis/analysis/msd.py [hunk 2]
```diff
@@ -270,14 +272,14 @@ class EinsteinMSD(AnalysisBase):
...
         r"""
         Parameters
         ----------
-        u : Universe
-            An MDAnalysis :class:`Universe`.
+        u : Universe or AtomGroup
+            An MDAnalysis :class:`Universe` or :class:`AtomGroup`.
         select : str
-            A selection string. Defaults to `None` in which case
+            A selection string. Defaults to `all` in which case
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply package/MDAnalysis/analysis/msd.py:282]: Should I unduplicate it? Up to you. :-)

File: package/MDAnalysis/analysis/msd.py [hunk 3]
```diff
@@ -288,9 +290,12 @@ def __init__(self, u, select, msd_type='xyz', fft=True, **kwargs):
             Defaults to ``True``.    
 
         """
+        if isinstance(u, groups.UpdatingAtomGroup):
+            raise TypeError("UpdatingAtomGroups are not valid for MSD "
+                            "computation")
         self.u = u
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/msd.py:296]: Is this actually stored anywhere but line 302? I would suggest cleaning up the attributes list and just doing self._atoms = u.select_atoms(select). 

Although with hindsight, I'd suggest naming that attribute `self._atomgroup` (like the density code) or even going for a "non-private" version like `self.ag` like we do in places like Dihedrals and Hole2.

Beyond the scope of this conversion, we really should "unify" the way we store different things. I think there's a reasonable case for expecting that things are consistently stored/named across AnalysisBase classes.

File: package/MDAnalysis/analysis/msd.py [hunk 4]
```diff
@@ -288,9 +290,12 @@ def __init__(self, u, select, msd_type='xyz', fft=True, **kwargs):
             Defaults to ``True``.    
 
         """
+        if isinstance(u, groups.UpdatingAtomGroup):
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/msd.py:293]: Nearly forgot, please do add a test for this. Might as well try to keep test coverage for MSD at 100% :)

File: testsuite/MDAnalysisTests/analysis/test_msd.py [hunk 5]
```diff
@@ -22,6 +22,7 @@
 #
 
 
+from MDAnalysis.analysis.msd import EinsteinMSD as MSD
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_msd.py:25]: I was wondering why this was working and then I realised that when writing that block_import test I had forgotten to take into account that tidynamics isn't imported until you actually run the class (my bad).

I guess technically you don't even need the import inside of `test_notidynamics`. You could also move the test below the `scope=module` fixtures if you think that'd be cleaner.

File: package/MDAnalysis/analysis/msd.py [hunk 6]
```diff
@@ -288,9 +290,12 @@ def __init__(self, u, select, msd_type='xyz', fft=True, **kwargs):
...
+            raise TypeError("""UpdatingAtomGroups are not valid for MSD
+         computation""")
```
[the remark this answers is not in the document]
Comment (lilyminium) [reply package/MDAnalysis/analysis/msd.py:297]: Still a newline, I recommend removing that

**Comment by lilyminium** [comment]
Could you please add this to CHANGELOG under "enhancements" -- I know MSD was added in 2.0.0 but this does affect anyone who might have used it up to now :-)

**Comment by lilyminium** [comment]
Thank you @hmacdope! :-D

**Comment by hmacdope** [comment]
no thank you @lilyminium @IAlibay :-D

**Pull request merged**